File: .dev/orchestrator/commands/new-feature.js
import fs from 'fs/promises'
import path from 'path'
import chalk from 'chalk'
import inquirer from 'inquirer'
import { fileURLToPath } from 'url'

const __dirname = path.dirname(fileURLToPath(import.meta.url))

function slugify(text) {
  return text
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, '-')
    .replace(/^-|-$/g, '')
}

export async function newFeature(name) {
  try {
    console.log(chalk.cyan('\n--- Zuri: New Feature Spec ---\n'))

    // Prompt for module, description, priority
    const answers = await inquirer.prompt([
      {
        type: 'list',
        name: 'module',
        message: 'Select module:',
        choices: [
          'core/inbox',
          'core/crm',
          'core/pos',
          'core/marketing',
          'core/tasks',
          'core/ai',
          'shared/inventory',
          'shared/procurement',
          'industry/culinary/courses',
          'industry/culinary/enrollment',
        ],
      },
      {
        type: 'input',
        name: 'description',
        message: 'Short description:',
        validate: (v) => (v.trim() ? true : 'Description is required'),
      },
      {
        type: 'list',
        name: 'priority',
        message: 'Priority:',
        choices: ['P0', 'P1', 'P2', 'P3'],
        default: 'P1',
      },
    ])

    // Read template
    const templatePath = path.join(__dirname, '..', 'templates', 'feature-spec.md')
    let template = await fs.readFile(templatePath, 'utf-8')

    const today = new Date().toISOString().slice(0, 10)
    const slug = slugify(name)

    // Replace placeholders
    template = template.replace(/\{\{NAME\}\}/g, name)
    template = template.replace(/\{\{DATE\}\}/g, today)
    template = template.replace(/\{\{module\}\}/g, answers.module)

    // Update priority in frontmatter
    template = template.replace(/priority: P1/, `priority: ${answers.priority}`)

    // Insert description into Summary section
    template = template.replace(
      '<!-- 2-3 ประโยค อธิบายว่า feature นี้ทำอะไร ทำไมต้องมี -->',
      `<!-- 2-3 ประโยค อธิบายว่า feature นี้ทำอะไร ทำไมต้องมี -->\n${answers.description}`
    )

    // Fix the module line in frontmatter (template has core/{{module}})
    template = template.replace(`core/${answers.module}`, answers.module)

    // Write feature spec
    const featuresDir = path.join(process.cwd(), 'docs', 'product', 'features')
    await fs.mkdir(featuresDir, { recursive: true })
    const featurePath = path.join(featuresDir, `${slug}.md`)
    await fs.writeFile(featurePath, template, 'utf-8')
    console.log(chalk.green(`  Created: docs/product/features/${slug}.md`))

    // Write flow skeleton
    const flowsDir = path.join(process.cwd(), 'docs', 'product', 'flows')
    await fs.mkdir(flowsDir, { recursive: true })
    const flowPath = path.join(flowsDir, `${slug}-flow.md`)
    const flowContent = `---
title: "${name} — Flow"
feature: "${slug}"
created: "${today}"
---

# ${name} — Flow

## User Flow

\`\`\`
[Start] --> [Step 1] --> [Step 2] --> [End]
\`\`\`

## Sequence Diagram

<!-- Add Mermaid or ASCII sequence diagram here -->

## Edge Cases

- ...
`
    await fs.writeFile(flowPath, flowContent, 'utf-8')
    console.log(chalk.green(`  Created: docs/product/flows/${slug}-flow.md`))

    console.log(
      chalk.yellow('\n  Feature spec created — review + approve before implement\n')
    )
  } catch (error) {
    console.error(chalk.red('[new-feature]'), error.message)
    process.exitCode = 1
  }
}

File: scripts/new-feature.py
#!/usr/bin/env python3
"""
Zuri: New Feature Spec Creator
Usage: python scripts/new-feature.py "Feature Name"
       python scripts/new-feature.py  (interactive)

Creates:
  docs/product/specs/FEAT-{SLUG}.md   — feature spec (DRAFT)
  docs/product/flows/{slug}-flow.md   — flow skeleton
"""

import sys
import re
from datetime import date
from pathlib import Path

ROOT = Path(__file__).parent.parent
SPECS_DIR = ROOT / "docs" / "product" / "specs"
FLOWS_DIR = ROOT / "docs" / "product" / "flows"
TEMPLATE = ROOT / ".dev" / "orchestrator" / "templates" / "feature-spec.md"

MODULES = [
    "core/crm",
    "core/inbox",
    "core/pos",
    "core/marketing",
    "core/dsb",
    "core/tasks",
    "shared/auth",
    "shared/ai",
    "shared/multi-tenant",
    "shared/notifications",
    "industry/culinary/enrollment",
    "industry/culinary/kitchen",
]


def slugify(text: str) -> str:
    return re.sub(r"[^a-z0-9]+", "-", text.lower()).strip("-")


def pick_module() -> str:
    print("\n  Select module:")
    for i, m in enumerate(MODULES, 1):
        print(f"    {i:2d}. {m}")
    while True:
        choice = input("\n  Number [1]: ").strip() or "1"
        if choice.isdigit() and 1 <= int(choice) <= len(MODULES):
            return MODULES[int(choice) - 1]
        print("  Invalid choice — enter a number from the list")


def pick_priority() -> str:
    choices = ["P0", "P1", "P2", "P3"]
    print("\n  Priority: " + " / ".join(choices))
    p = input("  [P1]: ").strip().upper() or "P1"
    return p if p in choices else "P1"


def main():
    name = " ".join(sys.argv[1:]).strip() if len(sys.argv) > 1 else ""
    if not name:
        name = input("Feature name: ").strip()
    if not name:
        print("Error: feature name required")
        sys.exit(1)

    slug = slugify(name)
    today = date.today().isoformat()

    description = input("  Short description: ").strip()
    module = pick_module()
    priority = pick_priority()

    SPECS_DIR.mkdir(parents=True, exist_ok=True)
    FLOWS_DIR.mkdir(parents=True, exist_ok=True)

    # --- Feature Spec ---
    if TEMPLATE.exists():
        template = TEMPLATE.read_text(encoding="utf-8")
        template = template.replace("{{NAME}}", slug.upper())
        template = template.replace("{{subtitle}}", description)
        template = template.replace("{{DATE}}", today)
        template = template.replace("{{module}}", module)
        if description:
            template = template.replace(
                "<!-- 2-3 ประโยค อธิบายว่า feature นี้ทำอะไร ทำไมต้องมี -->",
                f"<!-- 2-3 ประโยค อธิบายว่า feature นี้ทำอะไร ทำไมต้องมี -->\n{description}",
            )
    else:
        template = f"""---
title: "FEAT-{slug.upper()}: {name}"
module: {module}
priority: {priority}
status: DRAFT
created: {today}
---

# FEAT-{slug.upper()}: {name}

> {description}

## Overview

<!-- 2-3 ประโยค อธิบายว่า feature นี้ทำอะไร ทำไมต้องมี -->
{description}

## Requirements

### Functional
- ...

### Non-Functional
- ...

## Acceptance Criteria
- [ ] ...

## Out of Scope
- ...
"""

    spec_path = SPECS_DIR / f"FEAT-{slug.upper()}.md"
    spec_path.write_text(template, encoding="utf-8")
    print(f"\n  Created: docs/product/specs/FEAT-{slug.upper()}.md")

    # --- Flow Skeleton ---
    flow_content = f"""---
title: "{name} — Flow"
feature: "{slug}"
created: "{today}"
---

# {name} — Flow

## User Flow

```
[Start] --> [Step 1] --> [Step 2] --> [End]
```

## Sequence Diagram

<!-- Add Mermaid or ASCII sequence diagram here -->

## Edge Cases

- ...
"""
    flow_path = FLOWS_DIR / f"{slug}-flow.md"
    flow_path.write_text(flow_content, encoding="utf-8")
    print(f"  Created: docs/product/flows/{slug}-flow.md")

    print("\n  Feature spec created — review + get Boss approval before implement\n")


if __name__ == "__main__":
    main()
